specfile/specfile.py

```diff
@@ -111,6 +149,43 @@ def __exit__(
     ) -> None:
         self.save()
 
+    def __deepcopy__(self, memodict: Dict[int, Any]):
+        """
+        Deepcopies the object, handling file-like attributes.
+        """
+        specfile = self.__class__.__new__(self.__class__)
+        memodict[id(self)] = specfile
+
+        for k, v in self.__dict__.items():
+            if k == "_file":
+                continue
+            setattr(specfile, k, copy.deepcopy(v, memodict))
+
+        try:
+            path = Path(self._file.name)
```

Comment:
Here we are again treating `self._file` as named file.

```suggestion
            path = Path(cast(FileIO, self._file).name)
```